mmv1/third_party/terraform/tests/resource_storage_transfer_agent_pool_test.go

```diff
@@ -0,0 +1,196 @@
+package google
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccStorageTransferAgentPool_agentPoolUpdate(t *testing.T) {
+	t.Parallel()
+
+	agentPoolName := fmt.Sprintf("tf-test-agent-pool-%s", randString(t, 10))
+	displayName := fmt.Sprintf("tf-test-display-name-%s", randString(t, 10))
+	displayNameUpdate := fmt.Sprintf("tf-test-display-name-%s", randString(t, 10))
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckStorageTransferAgentPoolDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccStorageTransferAgentPool_agentPoolBasic(getTestProjectFromEnv(), agentPoolName, displayName),
+			},
+			{
+				ResourceName:            "google_storage_transfer_agent_pool.foo",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"state"},
```

Comment:
I guess we can remove this since we've waited on the state? Have tried it locally?